services/related.py

```
@@ -0,0 +1,364 @@
+                else:
+                    # No remote posts found - this could mean the blogs directory doesn't exist
+                    # or there are no published posts yet
+                    if repo:
+                        logger.info(f"No remote posts found in {remote_repo} - this may be a new repo or no blogs published yet")
+                        # When working on a PR, be conservative and don't include local posts
+                        # that might not exist on the target branch
+                        published_posts = []
+                    else:
+                        # No repo specified, safe to use local posts
+                        logger.info("No remote posts found, using local posts")
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)